atc/lidar/scanner_test.go

```diff
@@ -105,7 +105,7 @@ var _ = Describe("Scanner", func() {
 
 				Context("when the resource parent type is a base type", func() {
```

Comment:
Maybe we should separate out another context for testing checking of a resource using a custom resource type? That way we can keep this context to be strictly testing the checking of a resource that uses a base type and will only call `TryCreateCheck` once for the resource.

Reply:
That's actually handled further down: https://github.com/concourse/concourse/blob/f6033fb8f7f8da61361ae30df940369288dbcaab/atc/lidar/scanner_test.go#L208-L216

It turns out the reason this is doing multiple checks here is that base resource types shouldn't be returned by `ResourceTypes()` https://github.com/concourse/concourse/blob/f6033fb8f7f8da61361ae30df940369288dbcaab/atc/lidar/scanner_test.go#L103